### ISSUE TITLE
Potential Bug with TSQL Dialect QUOTED_IDENTIFIER keyword

### ISSUE THREAD

**Issue by edmoren** [sqlfluff/sqlfluff#2611]
## Expected Behaviour

Setting the value for `QUOTED_IDENTIFIER` should not trigger the unnecessary quoted_identifier rule L059 for the following tsql code.
```
SET ANSI_NULLS ON
GO

SET QUOTED_IDENTIFIER ON
GO






CREATE PROCEDURE [dbo].[usp_job_email_route_error]
	@in_debug INT = 0

AS

/****************************************************************************************************************************************************************************************

	Will send an email containing route error information

*****************************************************************************************************************************************************************************************
Changelog (User / Date / Comments):

****************************************************************************************************************************************************************************************/
```

## Observed Behaviour
Running `sqlfluff lint` on the attached `sample.txt` file will result in this output. 
```
$ sqlfluff lint ~/Desktop/sample.sql --config ~/source/repos/company/solution/database/.sqlfluff 
== [C:\Users\user\Desktop\sample.sql] FAIL
L:   4 | P:   5 | L059 | Unnecessary quoted identifier QUOTED_IDENTIFIER.
L:   5 | P:   3 |  PRS | Line 5, Position 3: Found unparsable section:
                       | '\n\n\n\n\n\n\nCREATE PROCEDURE [dbo].[usp_job_e...'
L:  24 | P:   1 | L009 | Files must end with a single trailing newline.
All Finished 📜 🎉!
```

Running `sqlfluff fix` will result in 'trimming' the keyword `QUOTED_IDENTIFIER`. 

I was getting a result that looks as such for line 4: `SET UOTED_IDENTIFIE ON`

## Steps to Reproduce
Save `sample.txt` to `.sql` file. 
Run `sqlfluff lint <file_name>`
Check line 4. Line 5 also showing parse error, but I am not convinced this is a bug.  

## Dialect
TSQL 

## Version
sqlfluff, version 0.10.0
Python 3.9.10

## Configuration
```
[sqlfluff]
dialect = tsql

[sqlfluff:indentation]
indented_joins = false
indented_using_on = true

[sqlfluff:rules]
indent_unit = tab
tab_space_size = 4
operator_new_lines = after
capitalisation_policy = upper
aliasing = explicit
allow_scalar = True
extended_capitalisation_policy = consistent
ignore_comment_clauses = True
ignore_comment_lines = True
max_line_length = 120
comma_style = leading
single_table_references = consistent
select_clause_trailing_comma = forbid
prefer_count_1 = True
multiline_newline = True
require_final_semicolon = True
allow_space_in_identifier = True
quoted_identifiers_policy = all
unquoted_identifiers_policy = all
exclude_rules = L025, L031, L061
```
[parse.txt](https://github.com/sqlfluff/sqlfluff/files/8048142/parse.txt)
[sample.txt](https://github.com/sqlfluff/sqlfluff/files/8048146/sample.txt)

**Comment by tunetheweb** [comment]
I can't repeat the L059 error?

```
% sqlfluff lint test.sql               
== [test.sql] FAIL                                                                                                                                                                                        
L:   1 | P:  19 |  PRS | Line 1, Position 19: Found unparsable section: '\nCREATE
                       | TABLE IF NOT EXISTS "tbl" ( "col...'
All Finished 📜 🎉!
```

**Comment by tunetheweb** [comment]
Sorry ignore me. User error. Looking...

**Comment by tunetheweb** [comment]
Fixed in #2614

**Comment by edmoren** [comment]
@tunetheweb Thanks. Has this latest version already been pushed up to `pip` or will I need to do a manual install from the repo?

**Comment by tunetheweb** [comment]
Fix hasn’t been merged yet but even after that we typically do releases every few weeks.

You should be able to `--noqa: L059` that line in the meantime.

**Comment by edmoren** [comment]
Cool cool. Thanks for the quick fix.
